vochain/state.go

```diff
@@ -58,15 +184,21 @@ func (e ErrHaltVochain) Unwrap() error { return e.reason }
 
 // State represents the state of the vochain application
 type State struct {
-	Store     statedb.StateDB
-	voteCache *lru.Cache
+	Store             *statedb.StateDB
+	Tx                *statedb.TreeTx
+	mainTreeViewValue atomic.Value
+	voteCache         *lru.Cache
 	ImmutableState
 	mempoolRemoveTxKeys func([][32]byte, bool)
 	txCounter           int32
 	eventListeners      []EventListener
 	height              uint32
 }
 
+// NOTE(Edu): It is unclear to me why we use locking for write operations here.
```

Comment:
Agree, also, the line 204 comment makes reference to IAVL trees, which are no longer there.

Reply:
Right. I've updated the comment, stating as a TODO to remove the mutex once we're ready to do proper integration tests.